### ISSUE TITLE
Create example language templates for flox init

### ISSUE THREAD

**Issue by samrose** [flox/floxpkgs#25]
related to https://github.com/flox/product/issues/16

**Comment by samrose** [comment]
pr for docs https://github.com/flox/floxdocs/pull/124

pr for elixir (merged) https://github.com/flox/floxpkgs/pull/40/files

pr for perl https://github.com/flox/floxpkgs/pull/44

pr for javascript https://github.com/flox/floxpkgs/pull/43

pr for java https://github.com/flox/floxpkgs/pull/42

**Comment by limeytexan** [comment]
@tomberek we've discussed many times the goal of reducing the boilerplate for these templates to the bare minimum; can we do anything by way of the flox CLI to provide the flake part dynamically so that these templates (and correspondingly the users' repositories) won't have to contain flake.{nix,lock} files? I just worry as we start getting people using the system that we'll end up with a ton of mess to clean up someday.

**Comment by robinbrantley** [comment]
Sam - Just an FYI that the user story (https://github.com/flox/product/issues/26) Michael created for this has a few more listed for you (just highlighting in case you weren't aware). I have linked this task to that story and each related PR in it at his request. If any of these are covered by another task already, just let me know and I'll update the links. :-)
- go
- rust
- jupyter
- C
- basel
- ninja

**Comment by samrose** [comment]
@limeytexan @robinbrantley please see the following PR

https://github.com/flox/floxdocs/pull/138


https://github.com/flox/floxpkgs/pull/50  and this one https://github.com/flox/floxpkgs/pull/43

I recommend merging 43, then 50 then the docs

I am working now on a bazel template. There is now builder for jupyter as it is covered under `buildPythonPackage` 

I can refactor the jupyter env template I created previously to be part of our templates directory, but it uses `buildPythonPackage`  and `mkDerivation` which are already covered. Please advise about what you have in mind for jupyter and how it fits with the pattern of the rest of the templates we've created (seems to diverge).

I am not familiar with ninja. Can you point me to what you refer to there?

**Comment by samrose** [comment]
@limeytexan @robinbrantley it seems like `jupyter` and `ninja` are more at home in `flox-examples` than these templates, which otherwise seem to be targeted toward system builders?

**Comment by samrose** [comment]
> @tomberek we've discussed many times the goal of reducing the boilerplate for these templates to the bare minimum; can we do anything by way of the flox CLI to provide the flake part dynamically so that these templates (and correspondingly the users' repositories) won't have to contain flake.{nix,lock} files? I just worry as we start getting people using the system that we'll end up with a ton of mess to clean up someday.

@limeytexan I really don't see any easy way to do this without creating some form of dynamic flake generation. Maybe @tomberek knows more than I do. But from what I understand this sounds like a large undertaking in order to get something that works effectively.

**Comment by robinbrantley** [comment]
@samrose - the related user story (https://github.com/flox/product/issues/26) has been updated to break down each of the builders needed (as some languages need more than 1 builder) and linked to the PR where the work was done. This is help clarify what is still missing.

I note that in that user story I separated and put both jupyter and ninja as a list under flox-examples as you suggested but note Michael just told me he thought it was a builder. I leave it to you to confirm either way. I can then create a new user story to flag those separately and leave just the builders in https://github.com/flox/product/issues/26 for simplicity.
